pof/src/types.rs

```diff
@@ -1325,8 +1326,10 @@ impl SubObject {
 }
 
 fn parse_uvec_fvec(props: &str) -> Option<(Vec3d, Vec3d)> {
-    let uvec = Vec3d::from_str(properties_get_field(props, "$uvec")?).ok()?;
-    let fvec = Vec3d::from_str(properties_get_field(props, "$fvec")?).ok()?;
+    // https://wiki.hard-light.net/index.php/Subobject_and_subsystem_properties#.24uvec:
+    // "It does not need to be a normalized vector."
+    let uvec = Vec3d::from_str(properties_get_field(props, "$uvec")?).ok()?.normalize();
+    let fvec = Vec3d::from_str(properties_get_field(props, "$fvec")?).ok()?.normalize();
```

Comment:
Not sure about this code. If it's going to be auto-normalized, it should probably also be auto-orthogonalized. Regarding "It does not need to be a normalized vector", we should presumably do the same thing that FSO does here: if it normalizes then we should too, if it doesn't normalize then we should not normalize and warn instead.

Reply:
[Here's the relevant code](https://github.com/scp-fs2open/fs2open.github.com/blob/fe44dda80967e8554e238738e4575a39bd9d843a/code/model/modelread.cpp#L2038-L2047). It's doing a full orthogonalize operation, which only causes errors if the vectors are zero (or zero after orthogonalizing, i.e. parallel), so there is no need to have a warning for non-orthogonal vectors - `fvec` just gets priority in that case.

Reply:
Well, the advantage to having a warning for a non-perpendicular uvec/fvec is that it gives the modder more control; They can fix their own mistakes before the engine "fixes" their mistakes for them, and the turret ends up rotating in a direction it isn't supposed to.

Reply:
> Also, automatically normalizing the vectors lets modders work with integers, which are less confusing than "real numbers".

Automatically orthogonalizing has the same property. For example you might pin the up vector to 0,1,0 and then point the fvec in various directions like 1,2,3 and we orthogonalize only for the purpose of drawing the arrows but don't change the user's text or the in memory representation. As long as we follow the same algorithm as FSO it's not even wrong to have significantly not orthogonal vectors as long as they aren't literally (anti-)parallel.

Reply:
I think the user should be in control of the auto-orthogonalization. That way, they can see what their uvec/fvec looks like auto-orthogonalized, and can make the desired adjustments without being confused by the auto-orthogonalization.

Reply:
Is that not what he suggests? What the user *sees* is the orthogonalized result, but we don't directly touch what they type in (or eventually write). This way there are no barriers between what they input and what they'll get, they will see the result immediately.

Reply:
And what if the user *wants to* see what the vectors would look like without auto-orthogonalization?

Reply:
You can't patent algorithms, only code, and we aren't reusing the code. The algorithm here is very simple, we just need to get straight which one takes priority in the orthogonalization.

Reply:
On second thought, I think I do agree with the original implementation. Normalize, yes, but do not orthgonalize. I think it's much easier for users to reason about changes to normalized direction vectors, than it is for them to reason about changes to *orthogonalized* vectors as well, when *neither* the direction *nor* magnitude are necessarily maintained, I think it will be far too difficult for the user to understand what their changes are doing.

What we should do is to have an 'orthogonalize' button to fix the warning for non-orthogonal vectors (since there isn't really good UI support for subobject properties yet, I am not going ask for this in this PR).

Partly also because I don't really like porting the FSO orthgonalize operation because it is quite cumbersome.